execution/withdrawals.md

```diff
@@ -0,0 +1,70 @@
+# Withdrawals
+
+Support validator withdrawals from the beacon chain to the EVM via a "system-level" operation type.
+
+## Motivation
+
+Provide a way for beacon chain withdrawals to enter into the EVM. Extends [ethereum/eip-4895](https://eips.ethereum.org/EIPS/eip-4895) but instead of minting native tokens calls an predefined contract which disperses a non-native tokens.
+
+## Specification
+
+| Name                  | Value                                        |
+| --------------------- | -------------------------------------------- |
+| `FORK_TIMESTAMP`      | `TBD`                                        |
+| `WITHDRAWAL_CONTRACT` | `TBD`                                        |
+| `SYSTEM_SENDER`       | `0xfffffffffffffffffffffffffffffffffffffffe` |
+
+Beginning with the execution timestamp `FORK_TIMESTAMP`, execution clients **MUST** introduce the following extensions to payload validation and processing:
+
+The following sections of [ethereum/eip-4895](https://eips.ethereum.org/EIPS/eip-4895#system-level-operation-withdrawal) are implemented un-altered
+
+- System-level operation: withdrawal
+- New field in the execution payload: withdrawals
+- New field in the execution payload header: withdrawals root
+- Execution payload validity
+
+**State transition**
+
+The `withdrawals` in an execution payload are processed **after** any user-level transactions are applied.
+
+For each `withdrawal` in the list of `execution_payload.withdrawals`, the implementation must construct and execute an EVM transaction as follows:
+
+```
+sender: SYSTEM_SENDER
```

Comment:
In POSDAO system transaction rules are:
1. gasLimit checks are disabled compared to block.GasLimit - block.GasUsed.

https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs#L204-L220

2. Caller balance and nonce checks are disabled. Nonce for system address is not incremented.

https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs#L256-L287

https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs#L468-L471

3. Fees are not added:

https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs#L421-L448

4. block.GasUsed is not incremented:

https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs#L482-L485